src/components/veNft/veNftStakingTokenRangesModal.tsx

```diff
@@ -1,22 +1,64 @@
 import { Trans } from '@lingui/macro'
-import { Modal } from 'antd'
+import { Col, Modal, Row, Image } from 'antd'
+import { V2ProjectContext } from 'contexts/v2/projectContext'
+import { useContext } from 'react'
+import { getVeNftBaseImage } from 'utils/v2/veNft'
 
-interface StakingTokenRangesModalProps {
+type StakingTokenRangesModalProps = {
   visible: boolean
+  tokenSymbolDisplayText: string
   onCancel: VoidFunction
 }
 
-const StakingTokenRangesModal = ({
+export default function StakingTokenRangesModal({
   visible,
+  tokenSymbolDisplayText,
   onCancel,
-}: StakingTokenRangesModalProps) => {
+}: StakingTokenRangesModalProps) {
+  const {
+    veNft: { baseImagesHash, variants },
+  } = useContext(V2ProjectContext)
+
   return (
-    <Modal visible={visible} onCancel={onCancel}>
-      <h2>
-        <Trans>Token Ranges</Trans>
-      </h2>
+    <Modal
```

Comment:
**note (non-blocking)** I think we can improve this UI in future. Works for now 👍